Gym/DQN/main.swift

```diff
@@ -0,0 +1,278 @@
+// Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#if canImport(PythonKit)
+    import PythonKit
+#else
+    import Python
+#endif
+import TensorFlow
+
+// Force unwrapping with `!` does not provide source location when unwrapping `nil`, so we instead
+// make a utility function for debuggability.
+fileprivate extension Optional {
+    func unwrapped(file: StaticString = #filePath, line: UInt = #line) -> Wrapped {
+        guard let unwrapped = self else {
+            fatalError("Value is nil", file: (file), line: line)
+        }
+        return unwrapped
+    }
+}
+
+// Initialize Python. This comment is a hook for internal use, do not remove.
+
+let np = Python.import("numpy")
+let gym = Python.import("gym")
+
+typealias State = Tensor<Float>
+typealias Action = Tensor<Int32>
+typealias Reward = Tensor<Float>
+
+class ReplayBuffer {
+    var states: Tensor<Float>
+    var actions: Tensor<Int32>
+    var rewards: Tensor<Float>
+    var nextStates: Tensor<Float>
+    let capacity: Int
+    var count: Int
+    var index: Int
```

Comment:
You can [set initial values for stored properties](https://docs.swift.org/swift-book/LanguageGuide/Initialization.html) `var count` and `var index` here, which is a bit clearer than initializing them in the initializer:

```suggestion
    var count: Int = 0
    var index: Int = 0
```

Reply:
Good point! Thank you for the suggestion.